ulimit.go

```diff
@@ -96,7 +96,11 @@ func ParseUlimit(val string) (*Ulimit, error) {
 		return nil, fmt.Errorf("too many limit value arguments - %s, can only have up to two, `soft[:hard]`", parts[1])
 	}
 
-	if soft > *hard {
+	if soft == -1 && *hard != -1 {
+		return nil, fmt.Errorf("ulimit soft limit must be less than or equal to hard limit: soft: -1 (unlimited), hard: %d", *hard)
+	}
+
+	if *hard != -1 && soft > *hard {
 		return nil, fmt.Errorf("ulimit soft limit must be less than or equal to hard limit: %d > %d", soft, *hard)
 	}
```

Comment:
LGTM, but may be the code can be written as 
```go
if *hard != -1 {
  if soft == -1 { ... }
  if soft > *hard { ... }
}
```

Reply:
Ah, was actually thinking about that after posting; I can update 👍